authenticator.go

```diff
@@ -22,16 +26,52 @@ type authenticator struct {
 	// `{{.Subject}}` is used by default if not set.
 	UserTemplate string
 
+	// GroupsClaimKey is the name of the key within the token claims that
+	// specifies which groups a user is a member of.
+	//
+	// `groups` is used by default if not set.
+	GroupsClaimKey string
+
+	// AuthorizedGroups is a list of groups required for authentication to pass.
+	// A user must be a member of at least one of the groups in the list, if
+	// specified.
+	//
+	// If the list is empty, group membership is not required for authentication
+	// to pass.
+	AuthorizedGroups []string
```

Comment:
Makes sense for us, I wonder if we should consider `CheckClaimKey` `CheckClaimValues` for wider appeal, or maybe json path key value pairs, or if the complexity around requirements would be just too much

Reply:
I think I'd like to defer it until we have more use cases, but I think you're right that if we see more use cases that require more one-off options like this, we should consider something bigger picture like JSON Path or even Rego.